ttnn/cpp/ttnn/operations/moreh/moreh_layer_norm_backward/device/moreh_layer_norm_backward_gamma_beta_grad_device_operation.cpp

```diff
@@ -0,0 +1,74 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "moreh_layer_norm_backward_gamma_beta_grad_device_operation.hpp"
+
+#include <cstdint>
+
+#include "tt_dnn/op_library/moreh_helper_functions.hpp"
+#include "ttnn/tensor/tensor.hpp"
+
+namespace ttnn::operations::moreh::moreh_layer_norm_backward_gamma_beta_grad {
+void MorehLayerNormBackwardGammaBetaGradOperation::validate_inputs(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {}
+
+MorehLayerNormBackwardGammaBetaGradOperation::program_factory_t MorehLayerNormBackwardGammaBetaGradOperation::select_program_factory(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    return ProgramFactory{};
+}
+
+void MorehLayerNormBackwardGammaBetaGradOperation::validate_on_program_cache_miss(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    validate_inputs(operation_attributes, tensor_args);
+};
+
+void MorehLayerNormBackwardGammaBetaGradOperation::validate_on_program_cache_hit(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    validate_inputs(operation_attributes, tensor_args);
+};
+
+MorehLayerNormBackwardGammaBetaGradOperation::shape_return_value_t MorehLayerNormBackwardGammaBetaGradOperation::compute_output_shapes(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    TT_FATAL(false, "The compute_output_shapes function in MorehLayerNormBackwardGammaBetaGrad is not implemented.");
+    return {};
+};
+
+MorehLayerNormBackwardGammaBetaGradOperation::tensor_return_value_t MorehLayerNormBackwardGammaBetaGradOperation::create_output_tensors(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    if (tensor_args.gamma_grad.has_value() && tensor_args.beta_grad.has_value()) {
+        return {tensor_args.gamma_grad.value(), tensor_args.beta_grad.value()};
+    }
```

Comment:
It seems that the current implementation only handles the case where both gamma and beta are present.
Please implement the cases where only one of them is present as well.

Reply:
It seems like the old version didn't handle that case properly. How about addressing all the test cases in the new epic? @hschoi4448 @ngohoang34 
```C
std::vector<Tensor> MorehLayerNormBackwardGammaBetaGrad::create_output_tensors(
    const std::vector<Tensor>& input_tensors, const std::vector<std::optional<Tensor>>& output_tensors) const {
    if (output_tensors.at(0).has_value() && output_tensors.at(1).has_value()) {
        log_debug(LogOp, "{}:{} use output tensor", __func__, __LINE__);
        return {output_tensors.at(0).value(), output_tensors.at(1).value()};
    }

    TT_THROW("Create output tensor is not supported yet. Fix this after the #9552 issue is addressed.");
    return {};
}
```